src/ensembl/src/content/app/custom-download/components/checkbox-with-selects/CheckboxWithSelects.test.tsx

```diff
@@ -31,107 +32,117 @@ const createOption = (isSelected = false) => ({
+
+    userEvent.click(checkboxElement as HTMLElement);
+
+    expect(container.querySelectorAll('.select').length).toBe(1);
+
+    expect(container.querySelectorAll('.removeIconHolder').length).toBe(0);
   });
 
   it('displays the remove button next to the Select if an option is selected', () => {
-    wrapper = mount(
-      <CheckboxWithSelects
-        {...defaultProps}
-        selectedOptions={[defaultProps.options[0].value]}
-      />
-    );
-
-    expect(wrapper.find('.removeIconHolder').length).toBe(1);
+    const { container } = renderCheckboxWithSelects({
+      selectedOptions: [defaultProps.options[0].value]
+    });
+
+    expect(container.querySelectorAll('.removeIconHolder').length).toBe(1);
   });
 
   it('displays the Plus button when one option is selected', () => {
-    wrapper = mount(
-      <CheckboxWithSelects
-        {...defaultProps}
-        selectedOptions={[defaultProps.options[0].value]}
-      />
-    );
-
-    expect(wrapper.find('.addIconHolder').length).toBe(1);
+    const { container } = renderCheckboxWithSelects({
+      selectedOptions: [defaultProps.options[0].value]
+    });
+
+    expect(container.querySelectorAll('.addIconHolder').length).toBe(1);
   });
 
   it('displays another select when the plus button is clicked', () => {
-    wrapper = mount(
-      <CheckboxWithSelects
-        {...defaultProps}
-        selectedOptions={[defaultProps.options[0].value]}
-      />
-    );
+    const { container } = renderCheckboxWithSelects({
+      selectedOptions: [defaultProps.options[0].value]
+    });
+
+    expect(container.querySelectorAll('.select').length).toBe(1);
 
-    expect(wrapper.find(Select)).toHaveLength(1);
+    const addIcon = container.querySelector('.addIconHolder .imageButton');
 
-    wrapper.find('.addIconHolder').find(ImageButton).simulate('click');
+    userEvent.click(addIcon as HTMLElement);
 
-    expect(wrapper.find(Select)).toHaveLength(2);
+    expect(container.querySelectorAll('.select').length).toBe(2);
   });
 
   it('hides the options that are already selected within the new Select', () => {
-    wrapper = mount(
-      <CheckboxWithSelects
-        {...defaultProps}
-        selectedOptions={[defaultProps.options[0].value]}
-      />
-    );
-
-    wrapper.find('.addIconHolder').find(ImageButton).simulate('click');
-    expect(wrapper.find(Select).last().prop('options')).toHaveLength(4);
+    const { container } = renderCheckboxWithSelects({
+      selectedOptions: [defaultProps.options[0].value]
+    });
+
+    const addIcon = container.querySelector('.addIconHolder .imageButton');
+
+    userEvent.click(addIcon as HTMLElement);
+
+    const allSelects = container.querySelectorAll('.select');
+    const lastSelect = allSelects[allSelects.length - 1];
+    const lastSelectControl = lastSelect.querySelector('.selectControl');
+
+    userEvent.click(lastSelectControl as HTMLElement);
+
+    expect(lastSelect.querySelectorAll('.option')).toHaveLength(4);
```

Comment:
4 is a bit of a magic number here. Is it `defaultProps.options.length - 1`?

Also, are we checking that the select will show all options if none is selected?

Reply:
Changed it to `defaultProps.options.length - 1` and added another test